### PR TITLE
TS-4990: Support new apis in ts_lua.

### PR DESCRIPTION
**Pull request by shukitchan**
support new apis in ts_lua

@jpeach pls take a look and review. Thanks.

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1058/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/950/ for details.

**Comment by shukitchan** [comment]
@jpeach pls take a look if you get a chance. thanks.

File: plugins/experimental/ts_lua/ts_lua_http.c [hunk 1]
```diff
@@ -387,6 +411,52 @@ ts_lua_http_set_cache_lookup_url(lua_State *L)
...
+  int port = 0;
+  const char *target;
+
+  ts_lua_http_ctx *http_ctx;
+
+  GET_HTTP_CONTEXT(http_ctx, L);
+
+  n = lua_gettop(L);
+
+  if (n == 2) {
```
Comment (jpeach) [inline plugins/experimental/ts_lua/ts_lua_http.c]: Throw an error if `n != 2`?

File: plugins/experimental/ts_lua/ts_lua_http.c [hunk 2]
```diff
@@ -526,6 +596,69 @@ ts_lua_http_resp_cache_untransformed(lua_State *L)
...
+}
+
+static int
+ts_lua_http_is_websocket(lua_State *L)
+{
+  ts_lua_http_ctx *http_ctx;
+
+  GET_HTTP_CONTEXT(http_ctx, L);
+
+  if (TSHttpTxnIsWebsocket(http_ctx->txnp)) {
```
Comment (jpeach) [inline plugins/experimental/ts_lua/ts_lua_http.c]: Consider:
```C
lua_pushboolean(L, TSHttpTxnIsWebsocket(http_ctx->txnp));
```

File: plugins/experimental/ts_lua/ts_lua_server_request.c [hunk 3]
```diff
@@ -755,3 +765,46 @@ ts_lua_server_request_server_addr_get_addr(lua_State *L)
 
   return 3;
 }
+
+static int
+ts_lua_server_request_server_addr_set_addr(lua_State *L)
+{
+  struct sockaddr_in addr4;
+  struct sockaddr_in6 addr6;
```
Comment (jpeach) [inline plugins/experimental/ts_lua/ts_lua_server_request.c]:
```C
union {
    struct sockaddr_in sin4;
    struct sockaddr_in6 sin6;
    struct sockaddr sa;
} addr;
...

TSHttpTxnServerAddrSet(http_ctx->txnp, &addr.sa);

**Comment by shukitchan** [comment]
@jpeach, I have updated the PR according to the comments. Please take a look.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/1023/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1129/ for details.

File: doc/admin-guide/plugins/ts_lua.en.rst [hunk 4]
```diff
@@ -1424,6 +1460,38 @@ We will get the output:
...
+**description**: This function can be used to set socket address of the origin server.
+
+The ts.server_request.server_addr.set_addr function requires three inputs, ip is a string, port and family is number.
+
+Here is an example:
+
+::
+
+    function do_global_read_request()
+        ts.server_request.server_addr.set_addr("192,168,231,17", 80, TS_LUA_AF_INET)
```
Comment (jpeach) [inline doc/admin-guide/plugins/ts_lua.en.rst]: Commas?

**Comment by shukitchan** [comment]
thanks. will squash before merge.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1161/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1054/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1175/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1068/ for details.

**Pull request merged**